cloudinit/sources/DataSourceOVF.py

```diff
@@ -684,4 +740,73 @@ def _raise_error_status(prefix, error, event, config_file, conf):
     util.del_dir(os.path.dirname(config_file))
     raise error
 
+
+def load_cloudinit_data(metaPath, userPath):
+    """
+    Load the cloud-init meta data, user data, cfg and network from the
+    given files
+    """
+    LOG.debug('load meta data from: %s: user data from: %s',
+              metaPath, userPath)
+    md = {}
+    cfg = {}
+    ud = None
+    network = None
+
+    md = load(util.load_file(metaPath))
+
+    if 'network' in md:
+        network = md['network']
+        del md['network']
+
+    if userPath:
+        ud = util.load_file(userPath).replace("\r", "")
+    return md, ud, cfg, network
+
+
+def load(data):
+    '''
+    The meta data could be JSON or YAML. Since YAML is a strict superset of
+    JSON, we will unmarshal the data as YAML. If data is None then a new
+    dictionary is returned.
+    '''
+    if not data:
+        return {}
+    return safeyaml.load(data)
+
+
+def collect_imc_files(cust_conf):
+    '''
+    collect all the other imc files. Since these files are copied
+    before cust.cfg is copied, check if they exist or not directly.
+    - meta data:
+      mandatory if customization specification is raw cloud-init data
+    - user data:
+      optional if customization specification is raw cloud-init data
+    - nics.txt:
+      optional if customization specification is traditional one
+    '''
+    metaPath = None
+    userPath = None
+    nicsPath = None
+    metaDataFile = cust_conf.meta_data_name
+    if metaDataFile:
+        metaPath = os.path.join(VMWARE_IMC_DIR, metaDataFile)
+        if not os.path.exists(metaPath):
+            raise FileNotFoundError("meta data file is not found")
+
+        userDataFile = cust_conf.user_data_name
+        if userDataFile:
+            userPath = os.path.join(VMWARE_IMC_DIR, userDataFile)
+            if not os.path.exists(userPath):
+                raise FileNotFoundError("user data file is not found")
+    else:
+        nicsPath = os.path.join(VMWARE_IMC_DIR, "nics.txt")
+        if not os.path.exists(nicsPath):
+            LOG.debug('nics.txt is not exist.')
```

Comment:
Probably worth actually parameterizing these logs since your log messages redact the full path of metadata, userdat and nics.txt.
```suggestion
            LOG.debug('%s does not exist.', nics_path)
```

Reply:
Fixed, PTAL.